frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -724,19 +742,74 @@ export default function SurveyEdit(): JSX.Element {
                                                                       />
                                                                   )}
                                                               </div>
                                                           )}
                                                       </LemonField>
+                                                      {survey.linked_flag?.filters.multivariate && (
+                                                          <LemonField.Pure
+                                                              label="Link to a specific flag variant"
+                                                              info="Choose which variant of the feature flag to link to this survey."
+                                                          >
+                                                              <div className="flex flex-col gap-2">
+                                                                  <LemonSegmentedButton
+                                                                      className="min-w-1/3"
+                                                                      value={
+                                                                          survey.conditions?.linkedFlagVariant ??
+                                                                          ANY_VARIANT
+                                                                      }
+                                                                      options={variantOptions(
+                                                                          survey.linked_flag?.filters.multivariate ||
+                                                                              undefined
+                                                                      )}
+                                                                      onChange={(variant) => {
+                                                                          setSurveyValue('conditions', {
+                                                                              ...survey.conditions,
+                                                                              linkedFlagVariant:
+                                                                                  variant === ANY_VARIANT
+                                                                                      ? null
+                                                                                      : variant,
+                                                                          })
+                                                                      }}
+                                                                  />
+                                                                  <p className="text-sm text-secondary">
+                                                                      This is a multi-variant flag. You can link to
```

Comment:
i'll need to add the min sdk version note before merging, still have to do the js changes now